fix(deps): adjust dependencies required for grails 7

jamesfredley opened the pull request:
These changes are currently required to start a grails 7 web project.

The `graal-sdk` dependency will be handled as follows:  https://github.com/bertramdev/asset-pipeline/pull/352#issuecomment-2407459979

After removing the asset-pipeline Gradle plugin from `buildSrc/build.gradle` and `setting.gradle` and defining it in one location `build.gradle`, the `graal-sdk` dependency is no longer needed.

```
plugins {
     id "com.bertramlabs.asset-pipeline" version "5.0.1"
}
```

The larger consolidation of buildSrc, settings.gradle and build.gradle will occur on https://github.com/grails/grails-forge/issues/347

jamesfredley committed:
add dependencies required for grails 7 web project to start

jamesfredley committed:
use java toolchain

codeconsole reviewed (dismissed):
I don't need that dependency to start?

jamesfredley commented:
@codeconsole which one?  The demo app uses sitemesh layouts.

jamesfredley committed:
remove need for graal-sdk dependency by defining Gradle asset-pipeline plugin in build.gradle only

implementation("com.bertramlabs.plugins:asset-pipeline-gradle:5.0.1") in buildSrc/build.gradle was requiring the extra graal-sdk dependency

codeconsole commented:
I am not sure `grails-plugin-sitemesh3` should be added as a dependency.

jamesfredley commented:
Once that is in place, we will remove it here.  

7.0.0-SNAPSHOT is currently an option on start.grails.org and this enables the generated application to start, for now.

codeconsole commented:
https://github.com/grails/grails-forge/issues/399
